core/src/main/java/com/oath/oak/common/OakCommonFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.oath.oak.common;
+
+import com.oath.oak.OakComparator;
+import com.oath.oak.OakMapBuilder;
+import com.oath.oak.OakSerializer;
+import com.oath.oak.common.intbuffer.OakIntBufferComparator;
+import com.oath.oak.common.intbuffer.OakIntBufferSerializer;
+import com.oath.oak.common.integer.OakIntComparator;
+import com.oath.oak.common.integer.OakIntSerializer;
+import com.oath.oak.common.string.OakStringComparator;
+import com.oath.oak.common.string.OakStringSerializer;
+
+import java.nio.ByteBuffer;
+import java.util.Comparator;
+
+public class OakCommonFactory {
```

Comment:
1) This is something to be used by external user as well, right? Should he get the "normal" builder from here as well? Though this is only for "specific" builders...?
2) OakCommonFactory --> at least OakMapCommonFactory, but still it is confusing because it sounds like something creating "Oaks" and not OakMapBuilders... Bbuilder is also kind of factory. May be OakMapBuilderFactory? Then coming to my first question from where to take the "normal/unspecific" OakMapBuilder...?
3) This change should be better reviewed by Eshcar as well. She is better in software engineering staff... But generally looks good to me.

Reply:
From what I understand this is a combination of 2 design patterns Builder pattern and Factory pattern - Liran correct me if I'm wrong:
The Builder pattern already existed before this PR (I believe it was introduced either by EranM or Anastasia); it allows creating an OakMap object and adding granularity to the object by applying chains of method calls to builder. This way the c'tor of the builder only gets mandatory attributes and the rest are set by the chain. Usually the Builder itself is a static class, and its methods return a builder object this allows the chain of method calls. Finally to get the OakMap object you call the build method, for example
`OakMap<Integer, Integer> oak = builder.build();` 
To be able to create different types of OakMap objects (<int, int>, <string,string>, etc.) you need different builder objects.
This is where the Factory pattern comes in (you may  want to change the name to OakMapBuilderFactory if this is indeed the intent). 
The factory is responsible for generating default builders for a predefined set of OakMap types.  For creating these builders the ctor gets the 4 mandatory parameters that are required for building an OakMap object. 
The 4 parameters include comparator and serializer. These are captured in the new classes like OakIntComparator that are also added in this PR.
I assume the main idea is to have it all in one place and not spreader around in the test classes. And perhaps this is also beneficial for the application.

Did I get this all correct?

Reply:
I assume Oak library might include non-map data-structures in the future. So maybe: `OakCommonBuildersFactory`?

Anastasia:
1.a. Used by external users.
1.b. No. It gives the user common builders, but not replace the main builder.

Eshcar: You got everything right.